src/main/cpp/faultinj/CMakeLists.txt

```diff
@@ -36,6 +36,8 @@ include_directories(
   "${SPDLOG_INCLUDE}"
 )
 
-target_link_libraries(
-  cufaultinj CUDA::cupti_static
-)
+if (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "aarch64")
```

Comment:
Whereas the first use case for dynamic linking may be a particular architecture aarch64 but I would just make it a boolean option that can be passed on the CLI to the tune of -DCUPTI_STATIC=ON